File: src/main/java/com/bdev/hogwarts_api/service/group/GroupService.java
package com.bdev.hogwarts_api.service.group;

import com.bdev.hogwarts_api.data.dto.group.Group;

import java.util.List;
import java.util.Optional;

public interface GroupService {
    boolean exists(long id);
    List<Group> getAllGroups();
    Optional<Group> getGroup(long id);
    long createGroup(Group group);
    void editGroup(Group group);
    void deleteGroup(long id);
}

File: src/main/java/com/bdev/hogwarts_api/service/group/GroupService.kt
package com.bdev.hogwarts_api.service.group

import com.bdev.hogwarts_api.data.dto.group.Group

interface GroupService {
    fun getAllGroups(): List<Group>
    fun exists(id: Long): Boolean
    fun getGroup(id: Long): Group?
    fun createGroup(group: Group): Long
    fun editGroup(group: Group)
    fun deleteGroup(id: Long)
}
